ts/torch_handler/base_handler.py

```diff
@@ -44,14 +44,12 @@ def initialize(self, context):
 
         """
         properties = context.system_properties
-        self.map_location = "cuda" if torch.cuda.is_available() else "cpu"
-        self.device = torch.device(
-            self.map_location + ":" + str(properties.get("gpu_id"))
-            if torch.cuda.is_available()
-            else self.map_location
-        )
-        self.manifest = context.manifest
+        self.device = "cuda" if torch.cuda.is_available() else "cpu"
```

Comment:
setting self.device to just "cuda" without id is suspect

Reply:
@ngimel Thanks for your comments, in fact the device was cuda+id,  
```
self.device = torch.device(
            self.map_location + ":" + str(properties.get("gpu_id"))
            if torch.cuda.is_available()
            else self.map_location
        )
```
however, this way, setting the torch.jit.load(model_pt_path, map_location=self.device) was failing. In current fix, the device id is getting set using [torch.cuda.set_device(properties.get("gpu_id"))](https://github.com/pytorch/serve/blob/issue_1037/ts/torch_handler/base_handler.py#L49).

Reply:
In what way is it failing?

Reply:
Looks like `torch.load` ignores map location, but `torch.jit.load` correctly loads the model to the specified location:
```
In [1]: import torch

In [2]: m=torch.jit.load("alexnet.pt", map_location="cuda:1")

In [3]: p=m.parameters().__next__()

In [4]: p.device
Out[4]: device(type='cuda', index=1)

In [5]: torch.cuda.memory.list_gpu_processes(0)
Out[5]: 'GPU:0\nno processes are running'

In [6]: torch.cuda.memory.list_gpu_processes(1)
Out[6]: 'GPU:1\nprocess    2011967 uses     1113.000 MB GPU memory'

In [7]: m=torch.load("alexnet.pt", map_location="cuda:1")
/data/users/ngimel/pytorch/torch/serialization.py:601: UserWarning: 'torch.load' received a zip file that looks like a TorchScript archive dispatching to 'torch.jit.load' (call 'torch.jit.load' directly to silence this warning)
  warnings.warn("'torch.load' received a zip file that looks like a TorchScript archive"

In [8]: p=m.parameters().__next__()

In [9]: p.device
Out[9]: device(type='cpu')

```
`torch.load` actually warns and suggests using `torch.jit.load`

Reply:
@ngimel, exactly, torch.load ignores the map_location, but jit.load works fine. The current workaround, using torch.cuda.set_device(gpu_id) seems working. So this might be a torchscripted checkpoint?

If we load the pre-trained model save the checkpoints then map_location works fine. 

```
from torchvision
import models
import torch
import time
from torchvision.models.alexnet import AlexNet
class ImageClassifier(AlexNet):
    def __init__(self):
        super(ImageClassifier, self).__init__()
              
device1='cuda:0'
device2='cuda:2'
model = models.alexnet(pretrained=True)
model2 = ImageClassifier()
model.eval()
model = model.to(device1)
torch.save(model.state_dict(),"alexnet_saved_cuda.pt")
state_dict = torch.load("alexnet_saved_cuda.pt", map_location=device2)
model2.load_state_dict(state_dict)
model2 = model2.to(device1)

```

Reply:
Cross linking https://github.com/pytorch/serve/pull/1038#issuecomment-831823980 -- I think this might be fixed with latest torchvision and was due to the fact that those checkpoints were saved with a very old version of PyTorch serializer

Reply:
Thanks @fmassa for the comments and updates, I have replaced the updated weights form torchvision master and adding the changes in this [PR](https://github.com/pytorch/serve/pull/1053).